generators/src/org/jetbrains/kotlin/generators/builtins/progressions.kt

```diff
@@ -47,6 +53,10 @@ class GenerateProgressions(out: PrintWriter) : BuiltInsSourceGenerator(out) {
             else -> throw IllegalArgumentException()
         }
 
+        val cast = when (kind) {
+            CHAR -> ".toInt()"
+            else -> ""
+        }
```

Comment:
The cast like `Int.toInt` is a no-op, so it isn't worth to complicate generator code to avoid it.

Reply:
I think it was needed a year ago, when I did the PR